src/components/pages/Gameification/GameWriteStep.js

```diff
@@ -0,0 +1,160 @@
+//** Import Modules */
+import React, { useEffect, useState } from 'react';
+import { gsap } from 'gsap';
+import { useHistory } from 'react-router-dom';
+import { Button } from 'antd';
+
+//** Import Components */
+import UploadDocs from '../../common/UploadDocs';
+
+//** Import Assets */
+import boyImg from '../../../assets/images/gamemodeimg/LightingKid.png';
+
+export default function GameWriteStep(props) {
+  // Get the history object
+  const history = useHistory();
+
+  // In this state, I save the amount of files we have ready to upload
+  const [fileList, setFileList] = useState([]);
+
+  // State to help when uploading
+  const [isUploading, setIsUploading] = useState(false);
+
+  // This handles what happens when a picture is added/removed
+  const handleChange = data => {
+    setFileList(data);
+  };
+
+  // This handles what happens when the files are uploaded and ready for submission
+  const handleSubmit = () => {
+    setIsUploading(true);
+
+    setTimeout(() => {
+      props.updateSubmissionData('HasWritten', true);
+      props.updateFileSubmissionData('writings', fileList);
+      props.updateBattleReady(true);
+
+      // Enable the modal window
+      const modalData = {
+        title: 'Get ready for the boss battle!',
+        description:
+          'To complete this mission, your work will be put up head-to-head against a boss. If you are ready to fight, submit your work!',
+        buttonTxt: "Let's Go!",
+      };
+
+      props.enableModalWindow(modalData);
+
+      setIsUploading(false);
+    }, 1500);
```

Comment:
Try to use constant variables for timeout values

Reply:
Got it

Reply:
Done.